tracks/go/exercises/pig-latin/mentoring.md

```diff
@@ -0,0 +1,72 @@
+# Pig Latin
+
+### Reasonable solutions:
+
+There is a (somewhat) reasonable solution using regular expressions, however most students are likely to use a combination of `strings` funtions or string slicing to solve this problem.
+
+Solutions vary considerably, but most reasonable solutions should demonstrate these three insights:
+
+1. Decoupled logic to break down a sentence into words and reform the Pig Latin sentence.
+
+```
+func Sentence(sentence string) string {
+	words := strings.Fields(sentence)
+	pigWords := make([]string, 0, len(words))
+	for _, w := range words {
```

Comment:
Are single-letter variables idiomatic Go code?

Reply:
What my seniors says is, it is idiomatic for short lived variables and loops (in small snippets as well).   However, for variables with broader scopes and long lifetimes, descriptive names are appreciated.

Reply:
[The greater the distance between a name's declaration and its uses, the longer the name should be](https://go.dev/talks/2014/names.slide#4). In this case, it is scoped to just a couple lines, so I consider this idiomatic.